### PR TITLE
Smbtorture update

### PR DESCRIPTION
**Pull request by spuiuk**
Use the newly added test_backend provided in the test-info.yml file to build the flapping lists for smbtorture tests.

### CODE REVIEW

**Comment by spuiuk** [comment]
I have pushed a flapping.xfs file with the failing tests. This has worked around the earlier issues.

However we still hit this problem. 
```
testcases/mount/test_mount.py::test_mount[192.168.123.10-gluster-vol-disperse] FAILED [  3%]
...
testcases/mount/test_mount.py::test_mount[192.168.123.11-gluster-vol-disperse] FAILED [  7%]
...
_______________ test_mount[192.168.123.10-gluster-vol-disperse] ________________

ipaddr = '192.168.123.10', share_name = 'gluster-vol-disperse'

    @pytest.mark.parametrize(
        "ipaddr,share_name", generate_mount_check(os.getenv("TEST_INFO_FILE"))
    )
    def test_mount(ipaddr: str, share_name: str) -> None:
>       mount_check(ipaddr, share_name)

testcases/mount/test_mount.py:53: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
testcases/mount/test_mount.py:27: in mount_check
    check_io_consistency(mount_point)
testcases/mount/mount_io.py:154: in check_io_consistency
    _check_io_consistency(rootdir)
testcases/mount/mount_io.py:116: in _check_io_consistency
    _run_checks(_make_datasets(base, 1024, 100))
testcases/mount/mount_io.py:94: in _run_checks
    dset.overwrite()
testcases/mount/mount_io.py:28: in overwrite
    self.write()
testcases/mount/mount_io.py:24: in write
    self.path.write_bytes(self.data)
/usr/lib64/python3.6/pathlib.py:1205: in write_bytes
    with self.open(mode='wb') as f:
/usr/lib64/python3.6/pathlib.py:1183: in open
    opener=self._opener)
/usr/lib64/python3.6/pathlib.py:1037: in _opener
    return self._accessor.open(self, flags, mode)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

pathobj = PosixPath('/tmp/56938/mnt_0/test_io_consistency/0')
args = (524865, 438)

    @functools.wraps(strfunc)
    def wrapped(pathobj, *args):
>       return strfunc(str(pathobj), *args)
E       OSError: [Errno 9] Bad file descriptor: '/tmp/56938/mnt_0/test_io_consistency/0'

/usr/lib64/python3.6/pathlib.py:387: OSError
----------------------------- Captured stdout call -----------------------------


Error while executing test_io_consistency: %s [Errno 9] Bad file descriptor: '/tmp/56938/mnt_0/test_io_consistency/0'
```

This is similar to what was reported earlier at
https://github.com/samba-in-kubernetes/sit-test-cases/pull/15#issuecomment-1661798943

The issue is unrelated to the changes we made here in this PR.

**Review by anoopcs9 — APPROVED** [review]

File: testcases/smbtorture/test_smbtorture.py [hunk 1]
```diff
@@ -48,7 +48,12 @@ def smbtorture(share_name: str, test: str, tmp_output: str) -> bool:
         filter_subunit_cmd.append(
             "--expected-failures=" + script_root + "/selftest/" + filter
         )
-    for filter in ["flapping", "flapping.d", "flapping.gluster"]:
+    flapping_list = ["flapping", "flapping.d"]
+    if "test_backend" in test_info and os.path.exists(
+        script_root + "/selftest/" + "flapping." + test_info["test_backend"]
+    ):
+        flapping_list.append("flapping." + test_info["test_backend"])
```
Comment (synarete) [inline testcases/smbtorture/test_smbtorture.py:55]: I prefer using `os.path.join` when constructing path-names. Also, consider using explicit temp-variables to make code clearer:

```
    flapping_name = test_info.get("test_backend", "xxxxxxx")
    flapping_path = os.path.join(script_root, "selftest", flapping_name)
    if os.path.exists(flapping_path):
        flapping_list.append(flapping_name)
    for filter in flapping_list:
        filter_subunit_cmd.append(
            "--flapping=" + os.path.join(script_root, "selftest", filter)
        )
```

**Review by synarete — COMMENTED** [review]
One minor style comment. Other than that, LGTM.

**Comment by spuiuk** [comment]
Retest this please

**Review by anoopcs9 — APPROVED** [review]

**Comment by anoopcs9** [comment]
@spuiuk 
> ./testcases/smbtorture/test_smbtorture.py:53:21: E711 comparison to None should be 'if cond is not None:'
./testcases/smbtorture/test_smbtorture.py:55:80: E501 line too long (81 > 79 characters)

**Comment by spuiuk** [comment]
The failure for glusterfs is at
```
test: samba3.smb2.create.mkdir-dup
time: 2023-08-22 12:16:18.052738Z
Testing SMB2 Create Directory with multiple connections
waiting for replies
time: 2023-08-22 12:17:18.291867Z
failure: samba3.smb2.create.mkdir-dup [
Exception: ../../source4/torture/smb2/create.c:1639: File 0 returned status NT_STATUS_IO_TIMEOUT
]
```

This has also been seen on and off before and was an existing issue.

**Review by synarete — APPROVED** [review]
LGTM

**Pull request merged**